### ISSUE TITLE
[Bug] post hook wrong declared

### ISSUE THREAD

**Issue by hasezoey** [szokodiakos/typegoose#314]
when doing something like this for the `pre` hook, everything works, no error
```ts
@pre<T>("save", hookwrapper)
@pre<T>("findOneAndUpdate", hookwrapper)
```
but when doing it with `post` hook
```ts
@post<T>("save", hookwrapper)
@post<T>("findOneAndUpdate", hookwrapper)
```
errors come:
`error TS2345: Argument of type '"save"' is not assignable to parameter of type '"insertMany"'`
and
`error TS2345: Argument of type '"findOneAndUpdate"' is not assignable to parameter of type '"insertMany"'`

as a Note, the wrapper function is:
```ts
function hookwrapper(...args: any[]) {
    return args.find((x) => typeof x === "function")(); // just to call next on any hook
}
```

the current `hooks.d.ts` file contains the following
```ts
declare type SingleMethod = 'findOne' | 'findOneAndRemove' | 'findOneAndUpdate' | DocumentMethod;
declare type MultipleMethod = 'find' | 'update';
export declare const pre: {
    <T>(method: "init" | "validate" | "save" | "remove", fn: DocumentPreSerialFn<T>): ClassDecorator;
    <T>(method: "init" | "validate" | "save" | "remove", parallel: boolean, fn: DocumentPreParallelFn<T>): ClassDecorator;
    <T>(method: "count" | "find" | "findOne" | "findOneAndRemove" | "findOneAndUpdate" | "update" | "insertMany" | "updateOne" | "updateMany", fn: SimplePreSerialFn<T>): ClassDecorator;
    <T>(method: "count" | "find" | "findOne" | "findOneAndRemove" | "findOneAndUpdate" | "update" | "insertMany" | "updateOne" | "updateMany", parallel: boolean, fn: SimplePreParallelFn<T>): ClassDecorator;
};
export declare const post: {
    <T>(method: "count", fn: PostNumberResponse<T>): ClassDecorator;
    <T>(method: "count", fn: PostNumberWithError<T>): ClassDecorator;
    <T>(method: SingleMethod, fn: PostSingleResponse<T>): ClassDecorator;
    <T>(method: SingleMethod, fn: PostSingleWithError<T>): ClassDecorator;
    <T>(method: MultipleMethod, fn: PostMultipleResponse<T>): ClassDecorator;
    <T>(method: MultipleMethod, fn: PostMultipleWithError<T>): ClassDecorator;
    <T>(method: "insertMany", fn: ModelPostFn<T> | PostMultipleResponse<T>): ClassDecorator;
};
```
the pre hook compiles right, but the post hook not

and some ending Notes:
- vscode Intellisense shows all available (including `save`) as working but when setting them, ts compile error happens
- typegoose 5.7.2

(already mentioned in #313)

**Comment by thurft** [comment]
can you update the information using `@post<T>("findOneAndUpdate", hookwrapper)` like in `@pre<T>("save"` 

i used `@pre<T>("findOneAndUpdate", (next: HookNextFunction, offer: any)` . but i dont have guidance on how to update the `offer` for validations or run things inside

**Comment by hasezoey** [comment]
> can you update the information using @post<T>("findOneAndUpdate", hookwrapper) like in @pre<T>("save"

sorry, but i dont understand what you mean, but i will update my issue with the wrapper function

**Comment by hasezoey** [comment]
Ok found the Issue, 
```ts
function hookwrapper(t: any, next: () => void) {
    return next();
}
```
did not have any before "next", so it tried to cast to "insertMany" (i dont know why)
-> adding anything before "next" fixed it

tl;dr: typescript has not given the "right" error
